docs/lua-modules/cohelper.md

```diff
@@ -0,0 +1,69 @@
+# cohelper Module
+| Since  | Origin / Contributor  | Maintainer  | Source  |
+| :----- | :-------------------- | :---------- | :------ |
+| 2019-07-24 | [TerryE](https://github.com/TerryE) | [TerryE](https://github.com/TerryE)  | [cohelper.lua](../../lua_modules/cohelper/cohelper.lua) |
+
+This module provides a simple wrapper around long running functions to allow 
+these to execute within the SDK and its advised limit of 15 mSec per individual
+task execution.  It does this by exploiting the standard Lua coroutine 
+functionality as described in the [Lua RM §2.11](https://www.lua.org/manual/5.1/manual.html#2.11) and [PiL Chapter 9](https://www.lua.org/pil/9.html).
+
+The NodeMCU Lua VM fully supports the standard coroutine functionality. Any
+interactive or callback tasks are executed in the default thread, and the coroutine
+itself runs in a second Lua thread. The coroutine can call any library functions,
+but any subsequent callbacks will, of course, execute in the default thread.
```

Comment:
There's still only actually one thread in Lua at a time, right?  I'm not sure that this discussion of threading deserves to be so fronted; it's worth keeping, but perhaps later in the document under `## Threading details` and with a disclaimer that it doesn't usually matter?

Reply:
No there are two definitely two separate Lua threads.  `L` is different for API calls from the these.  But this is standard Lua and explained in the Lua references that I provided.  Not withstanding this just like `node.js` the basic model is not pre-emptive. Execution of a thread is determined by its life or up to a yield / resume exchange in the case of coroutines